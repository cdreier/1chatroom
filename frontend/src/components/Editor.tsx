import React, { useContext } from 'react'
import { DemoStore } from '../store'
import { observer } from 'mobx-react-lite'

const Editor: React.FC = () => {
  const demo = useContext(DemoStore)
  return (
    <input value={demo.demostring} onChange={e => demo.demostring = e.target.value} data-testid="editorinput" />
  )
}

export default observer(Editor)
