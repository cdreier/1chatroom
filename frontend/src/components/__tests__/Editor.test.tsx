import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import Editor from '../Editor'

describe('testing the editor', () => {

  test('testing snapshot', () => {
    const component = renderer.create(
      <Editor />,
    ).toJSON()
    expect(component).toMatchSnapshot()
  })

  test('that keystrokes are working', () => {
    const { getByTestId } = render(
      <Editor />,
    )
    const element = getByTestId('editorinput')
    expect(element).toBeInTheDocument()
  })

})
