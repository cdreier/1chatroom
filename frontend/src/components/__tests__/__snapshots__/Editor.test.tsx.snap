// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the editor testing snapshot 1`] = `
<input
  data-testid="editorinput"
  onChange={[Function]}
  value="initial value"
/>
`;
